### PR TITLE
Fix Display Priority : Show only priorities used in matrix

### PR DESCRIPTION
**Pull request by tsmr**
Show only priorities used in matrix on ticket creation

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by orthagh — APPROVED** [review]
I though this existed before, but after a quick check, a limited setup of the matrix impacted the display of urgency and impact dropdowns.
So ok with the principle of altering priority dropdown also.

I pushed a mini fix as when testing the pr, i found the entry "Major" was missing on master

**Review by trasher — APPROVED** [review]

File: src/CommonITILObject.php [hunk 1]
```diff
@@ -2859,12 +2860,91 @@ public static function dropdownPriority(array $options = [])
...
+            }
+            if (
+                ($p['showtype'] == 'search')
+                || $CFG_GLPI[static::IMPACT_MASK_FIELD] & (1 << 1)
+            ) {
+                $impacts[] = 1;
+            }
+        }
+
+        $priorities = [];
```
Comment (cedric-anne) [inline src/CommonITILObject.php:2927]:
```suggestion
```

File: src/CommonITILObject.php [hunk 2]
```diff
@@ -2859,12 +2860,91 @@ public static function dropdownPriority(array $options = [])
...
+        $active_priorities = [];
+        foreach ($urgencies as $urgency) {
+            foreach ($impacts as $impact) {
+                if (isset($CFG_GLPI["_matrix_${urgency}_${impact}"])) {
+                    $priorities[] = $CFG_GLPI["_matrix_${urgency}_${impact}"];
```
Comment (cedric-anne) [inline src/CommonITILObject.php:2932]:
```suggestion
                    $active_priorities[] = $CFG_GLPI["_matrix_${urgency}_${impact}"];
```

File: src/CommonITILObject.php [hunk 3]
```diff
@@ -2859,12 +2860,91 @@ public static function dropdownPriority(array $options = [])
...
+                }
+            }
+        }
+        $active_priorities = array_unique($priorities);
```
Comment (cedric-anne) [inline src/CommonITILObject.php:2936]:
```suggestion
        $active_priorities = array_unique($active_priorities);
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**